packages/core/src/types.ts

```diff
@@ -19,12 +22,18 @@ export interface UserOperationCallData {
 
 export type BatchUserOperationCallData = UserOperationCallData[];
 
-export type UserOperationOverrides = Partial<
-  Pick<
-    UserOperationStruct,
-    "maxFeePerGas" | "maxPriorityFeePerGas" | "paymasterAndData"
-  >
->;
+export type UserOperationOverrides = Partial<{
```

Comment:
This is starting to leak alchemy specific schemas into `aa-core`. The way that we structured the overrides in `alchemy_requestGasAndPaymasterAndData` is non-standard. Is there a way to structure this such that our `aa-core` interfaces stay general, and we can apply this specific schema only in the alchemy provider?

Reply:
yea maybe we should make the methods that accept `overrides` fields in `aa-core` be generic on the overrides like:

`methodWithOverrides<O extends BaseOverrides>(uo: UoThingy, overrides: O)`

then in aa-alchemy we can make it like `methodWithOverrides<O extends BaseOverrides = AlchemyOverrides>(...)` wonder if it's possible to even make the signature be `methodWithOverrides(uo: UoThingy, overrides: AlchemyOverrides)` in aa-alchemy and have typescript infer the rest

Reply:
chatted offline some as well, but thought about it some more and we have had requests for "how do I add a buffer?" the current proposal is very annoying (you have to get a reference to the old feeEstimatorMiddleware and then write a new one that calls the old one + applies percent buffer). Introducing the percentage as an option in the base simplifies this.

With that it in mind @denniswon can you update the docs where we talk about adding a buffer (cc @avarobinson did we write this doc actually, or was that just a convo you had offline with a customer?)